lib/msf/core/payload/stager.rb

```diff
@@ -156,7 +156,14 @@ def handle_connection(conn, opts={})
     # If the stage should be sent over the client connection that is
     # established (which is the default), then go ahead and transmit it.
     if (stage_over_connection?)
-      p = generate_stage
+      uuid_raw = conn.get_once(16, 1)
+
+      opts = {}
+      if uuid_raw
+        opts[:uuid] = Msf::Payload::UUID.new({raw: uuid_raw})
```

Comment:
Same issue as above, this needs to only be called for UUID stagers and with a longer timeout.